ipaserver/install/server/upgrade.py

```diff
@@ -1700,6 +1702,34 @@ def migrate_to_authselect():
     sysupgrade.set_upgrade_state('authcfg', 'migrated_to_authselect', True)
 
 
+def fix_permissions():
+    """Fix permission of public accessible files and directories
+
+    In case IPA was installed with restricted umask, some public files and
+    directories may not be readable and accessible.
+
+    See https://pagure.io/freeipa/issue/7594
+    """
+    candidates = [
+        os.path.dirname(paths.GSSAPI_SESSION_KEY),
+        paths.CA_BUNDLE_PEM,
+        paths.KDC_CA_BUNDLE_PEM,
+        paths.IPA_CA_CRT,
+        paths.IPA_P11_KIT,
+    ]
+    for filename in candidates:
+        try:
+            s = os.stat(filename)
+        except OSError as e:
+            if e.errno != errno.ENOENT:
+                raise
```

Comment:
Please correct me if I am wrong,
this bare raise will raise an OSError exception. This exception is not caught anywhere.
In ipaserver/install/ipa_server_upgrade.py:52 is invocation of upgrade() in try block catching "only" RuntimeError.

Reply:
Yes, you are correct. It's intentional. The code block ignores ``ENOENT`` (no such file or directory), but will populate any other issue like file system or hard disk problems.

Reply:
PS: Yes, the function re-raises the exception, because any other case than ENOENT is an unrecoverable error. So any unrecoverable problem will bubble up and eventually cause the installer to fail. This will only happen when the system is totally broken.

Reply:
Yup this is what I meant thanks for explanation.